Userland/Libraries/LibSoftGPU/Buffer/Typed2DBuffer.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022, Jelle Raaijmakers <jelle@gmta.nl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/NonnullRefPtr.h>
+#include <AK/RefCounted.h>
+#include <AK/Try.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Rect.h>
+#include <LibGfx/Size.h>
+#include <LibSoftGPU/Buffer/Typed3DBuffer.h>
+
+namespace SoftGPU {
+
+/**
+ * Typed2DBuffer<T> wraps TypedBuffer<T> and only interacts on the 2D plane with z = 0.
+ */
+template<typename T>
+class Typed2DBuffer final : public RefCounted<Typed2DBuffer<T>> {
+public:
+    static ErrorOr<NonnullRefPtr<Typed2DBuffer>> try_create(Gfx::IntSize const& size)
+    {
+        auto buffer = TRY(Typed3DBuffer<T>::try_create(size.width(), size.height(), 1));
+        return adopt_ref(*new Typed2DBuffer(buffer));
+    }
+
+    void fill(T value, Gfx::IntRect const rect) { m_buffer->fill(value, rect.left(), rect.right(), rect.top(), rect.bottom(), 0, 0); }
```

Comment:
Is `Gfx::IntRect const rect` supposed to be a const reference?

Reply:
Made these `IntRect`s all const references :-) thanks!